enhancements/ingress/dynamic-config-manager.md

```diff
@@ -0,0 +1,469 @@
+
+Answer: **?**.
+
+We need to determine what kind of configuration changes cannot be implemented in
+DCM and either document these limitations or work with HAProxy upstream to
+enhance HAProxy's management capabilities and remove these limitations.
+
+#### Can we configure Dynamic Config Manager with 0 pre-allocated servers?
+
+Answer: **Probably no, but need verification**.
+
+One option could be to enable DCM initially with 0 pre-allocated server slots
+per backend.  If this worked, it would be useful to avoid reloads for scale-in,
+or for scale-out following scale-in.  However, we found that DCM still requires
+a reload for scale-in of a server that doesn't use a pre-allocated server slot.
+(TODO: This needs to be double-checked.)
+
+#### Is the cost of 1 pre-allocated server too high?
+
+Answer: **Probably no, but needs a release note**.
+
+Per https://gist.github.com/frobware/2b527ce3f040797909eff482a4776e0b, the cost
+of pre-allocating 1 server slot per backend is down to rounding error for 100
+routes, 4 to 9 MB for 1000 routes, and around 40 to 80 MB for 10000 routes.  The
+cost is highest if the router has many threads and many backends.  Most likely,
+if the router is configured with many threads and backends, it is already
+running on big machines.  However, it is important to call out the potential
+impact of DCM on memory consumption in a release note.
+
+#### Are any of the known defects blockers?
+
+Answer: **Probably yes**.
+
+Enabling Dynamic Config Manager should not cause any regressions.  We need to
+fix known issues regarding inaccurate metrics and session stickiness and any
+other regressions that we find
+(see [Risks and Mitigations](#risks-and-mitigations)).
+
+#### Do we need an override to turn off Dynamic Config Manager?
+
+Answer: **Probably yes**.
+
+Ingress is a critical cluster function, and it is an extremely
+performance-sensitive one for some customers.  In case we ship DCM enabled and a
+customer finds some critical issue, we need an unsupported config override that
+support staff can use to turn off DCM temporarily until we fix the issue.
+
+### Test Plan
+
+Dynamic Config Manager will be enabled initially using a TechPreviewNoUpgrade
+featuregate to provide CI signal.  We will additionally revisit existing E2E
+tests, enable any DCM-related E2E tests that are currently not enabled, and
+verify that all enabled router-related E2E tests pass with DCM.  As known
+defects are fixed and any unknown ones are discovered, we will add additional
+tests.  Finally, we will work with partners to verify DCM for their use-cases.
+
+Expanding test coverage is important both for verifying that DCM is ready for GA
+as well as for enabling us to improve DCM in future releases with confidence
+that we are not introducing regressions.
+
+### Graduation Criteria
+
+We will introduce the feature as TechPreviewNoUpgrade in OpenShift 4.18, with
+the goal of graduating it to GA in the same release.  Further improvements to
```

Comment:
Is the "goal of graduating it to GA in the same release" still applicable given our 2025 priorities?

Reply:
Depends on how things go with the Tech Preview, and whether we discover any new major issues.